provider_errors_test.go

```diff
@@ -0,0 +1,15 @@
+package goose
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/pressly/goose/v3/internal/check"
+)
+
+func TestPartialErrorUnwrap(t *testing.T) {
```

Comment:
Mind moving this to `provider_test.go`, so we can avoid creating a new file?

Otherwise, great idea!

Reply:
Done. Thanks for Goose!